server/src/main/java/io/druid/server/namespace/cache/LookupCoordinatorManager.java

```diff
@@ -0,0 +1,571 @@
+   */
+  public
+  @Nullable
+  Map<String, Object> getLookup(@NotNull final String tier, @NotNull final String lookupName)
+  {
+    final Map<String, Map<String, Map<String, Object>>> prior = getKnownLookups();
+    if (prior == null) {
+      LOG.warn("Requested tier [%s] lookupName [%s]. But no namespaces exist!", tier, lookupName);
+      return null;
+    }
+    final Map<String, Map<String, Object>> tierLookups = prior.get(tier);
+    if (tierLookups == null) {
+      LOG.warn("Tier [%s] does not exist", tier);
+      return null;
+    }
+    return tierLookups.get(lookupName);
+  }
+
+
+  @LifecycleStart
+  public void start()
+  {
+    synchronized (startStopSync) {
+      if (started) {
+        return;
+      }
+      if (executorService.isShutdown()) {
+        throw new ISE("Cannot restart after stop!");
+      }
+      lookupMapConfigRef = configManager.watch(
+          LOOKUP_CONFIG_KEY,
+          new TypeReference<Map<String, Map<String, Map<String, Object>>>>()
+          {
+          },
+          null
+      );
+      executorService.scheduleWithFixedDelay(
+          new Runnable()
+          {
+            @Override
+            public void run()
+            {
+              final Map<String, Map<String, Map<String, Object>>> allLookupTiers = lookupMapConfigRef.get();
+              // Sanity check for if we are shutting down
+              if (Thread.currentThread().isInterrupted()) {
+                LOG.info("Not updating namespace lookups because process was interrupted");
+                return;
+              }
+              if (!started) {
+                LOG.info("Not started. Returning");
+                return;
+              }
+              for (final String tier : allLookupTiers.keySet()) {
+                try {
+                  final Map<String, Map<String, Object>> allLookups = allLookupTiers.get(tier);
+                  final Map<String, Map<String, Object>> oldLookups = prior_update.get(tier);
+                  final Collection<String> drops;
+                  if (oldLookups == null) {
+                    drops = ImmutableList.of();
+                  } else {
+                    drops = Sets.difference(oldLookups.keySet(), allLookups.keySet());
+                  }
+                  if (allLookupTiers == prior_update) {
+                    LOG.debug("No updates");
+                    updateAllNewOnTier(tier, allLookups);
+                  } else {
+                    updateAllOnTier(tier, allLookups);
+                    deleteAllOnTier(tier, drops);
+                  }
+                }
+                catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
+                  throw Throwables.propagate(e);
+                }
+                catch (Exception e) {
+                  LOG.error(e, "Error updating namespaces for tier [%s]. Will try again soon", tier);
+                }
+              }
+              prior_update = allLookupTiers;
```

Comment:
In this iteration any tier:lookup_name combination under control of the coordinator is completely under its control. If people manually drop lookups from nodes this will try to repopulate them.

I think that's reasonable behavior.

The way for a node to have a lookup that is NOT under control of the coordinator would be to populate its value on the node outside of the coordinator cluster-wide config.